src/arrayMethodReduce.js

```diff
@@ -5,7 +5,20 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === 0
+          && initialValue === undefined
+          && arguments.length === 1) {
+        accumulator = this[0];
+        continue;
+      }
```

Comment:
This check will work on each loop but you really need id only once at the start. Maybe will be better to move it outside of your for loop.